web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -890,18 +892,27 @@ $.extend(fixmystreet.set_up, {
                 fixmystreet.map.setCenter(
                     marker.geometry.getBounds().getCenterLonLat(),
                     fixmystreet.map.getNumZoomLevels() - 1 );
+                history.pushState({
+                    reportId: reportId,
+                    reportPageUrl: reportPageUrl,
+                    mapState: fixmystreet.maps.get_map_state()
+                }, null);
             }
             return;
         }
 
+        if (fixmystreet.page === 'reports' || fixmystreet.page === 'around') {
+            around_map_state = fixmystreet.maps.get_map_state();
+        }
         fixmystreet.display.report(reportPageUrl, reportId, function() {
             // Since this navigation was the result of a user action,
             // we want to record the navigation as a state, so the user
             // can return to it later using their Back button.
             if ('pushState' in history) {
                 history.pushState({
                     reportId: reportId,
-                    reportPageUrl: reportPageUrl
+                    reportPageUrl: reportPageUrl,
+                    mapState: around_map_state
```

Comment:
I think this should be fixmystreet.maps.get_map_state() so that further pin clicks don't go back to the first map state.

Reply:
Yep, good catch.